### PR TITLE
Added Managed Node Groups

### PR DESCRIPTION
**Pull request by troy-ameigh**
Issue #74

*Description of changes:*
Added the new Manged Node Groups into the QuickStart. Also made sure that if selected you can only deploy version 1.14 or higher. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: templates/amazon-eks-nodegroup.template.yaml [hunk 1]
```diff
@@ -305,6 +315,8 @@ Conditions:
   UseCustomAmi: !Not [ !Equals [ !Ref CustomAmiId, "" ] ]
   CleanupSGs:
     !Not [ !Equals [ !Ref CleanupSecurityGroupDependenciesLambdaArn, "" ] ]
+  EKSmngdndgrp: !Equals [ !Ref 'ManagedNodeGroup' , 'no' ]
+  EKSvermngdndgrp: !Equals [ !Ref 'ManagedNodeGroup', 'yes' ]
```
Comment (jaymccon) [inline templates/amazon-eks-nodegroup.template.yaml]: `EnableManagedNodeGroup`/`DisableManagedNodeGroup` would be clearer imho.

File: templates/amazon-eks-nodegroup.template.yaml [hunk 2]
```diff
@@ -901,8 +931,31 @@ Resources:
...
+        DesiredSize: !Ref NumberOfNodes
+        MaxSize: !If [ MaxNodes, !Ref MaxNumberOfNodes, !Ref NumberOfNodes ]
+      Labels:
+        Key: Name
+        Value: !Sub "${EKSControlPlane}-${NodeGroupName}-Node"
+        PropagateAtLaunch: true
+      Subnets:
+        - !Ref PrivateSubnet1ID
+        - !Ref PrivateSubnet2ID
+        - !Ref PrivateSubnet3ID
```
Comment (jaymccon) [inline templates/amazon-eks-nodegroup.template.yaml]: Does this work if we have a 2 subnet deployment ?

File: templates/amazon-eks-nodegroup.template.yaml [hunk 3]
```diff
@@ -901,8 +931,31 @@ Resources:
...
 Outputs:
   EKSNodeSecurityGroup:
     Value: !Ref NodeSecurityGroup
+  NodeInstanceRole:
+    Description: The node instance role
+    Value: !GetAtt NodeInstanceRole.Arn
   NodeAutoScalingGroup:
-    Value: !Ref NodeGroup
\ No newline at end of file
+    Value: !If [ EKSmngdndgrp, !Ref 'NodeGroup', !Ref 'EKSNodegroup' ]
```
Comment (jaymccon) [inline templates/amazon-eks-nodegroup.template.yaml]: Has this been tested with cluster autoscaler? (it uses this output to know which asg to scale)

**Comment by troy-ameigh** [comment]
closing PR as there was an issue with the fork and a new fork had to be opened. I will open a new PR.

**Pull request closed without merging**